src/extension.ts

```diff
@@ -1731,9 +1736,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
     }
 }
 
-async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+async function useNamespaceKubernetes(explorerNode?: explorer.KubernetesObject) {
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], 'get', undefined, (resource) => {
+            if (resource) {
+                const currentNS = kubectlUtils.currentNamespace(kubectl);
+                currentNS.then((ns) => {
+                    //resource will be of format <kind>/<name>, hence get the substring
```

Comment:
I'm not sure if we're consistent about this but I think we usually put a space between `//` and the comment body.

Reply:
ok